Components/Games/GameLobby.jsx

```diff
@@ -29,12 +29,17 @@ class GameLobby extends React.Component {
         this.onNewGameClick = this.onNewGameClick.bind(this);
         this.onModalHidden = this.onModalHidden.bind(this);
 
+        let savedFilter = localStorage.getItem('gameFilter');
+        if(savedFilter) {
+            savedFilter = JSON.parse(savedFilter);
+        }
+
         this.state = {
             gameState: GameState.None,
             filter: {
-                beginner: true,
-                casual: true,
-                competitive: true
+                beginner: savedFilter ? savedFilter.beginner : true,
+                casual: savedFilter ? savedFilter.casual : true,
+                competitive: savedFilter ? savedFilter.competitive : true
```

Comment:
Super super minor thing, but there may be a bit more flexible approach if you have an object that has the default filters and then override it with the saved ones. i.e. 
```javascript
let filter = { beginner: true, casual: true, competitive: true };
let savedFilter = localStorage.getItem('gameFilter');
if(savedFilter) {
    filter = Object.assign(filter, JSON.parse(savedFilter);
}
```
This would have the advantage of automatically setting the default filter value for any new filters we add in the future. The current approach would need to add a check for `savedFilter.newField` being undefined and then setting it to the appropriate default value. Otherwise, the new field would always be unchecked to start.